### DISCUSSION TITLE
[feature request]: exec()

### DISCUSSION THREAD

**Discussion by nschloe** [wjakob/nanobind#156] [General]
### Problem description

[The documentation](https://nanobind.readthedocs.io/en/latest/porting.html) lists

> Removed features include:
> [...]
> ● Support for evaluating Python code strings was removed.
> [...]

I'd like to cast my vote for adding something like

```
nb:exec(code_as_string, nb::globals())
```

**Comment by wjakob** [comment]
(Cleaned up the history a bit after the conversion)

If this feature is important to you, my recommendation would be that you create a PR with a draft implementation or a port from pybind11 to receive feedback and iterate on the code until it can become part of the project.

**Reply by wjakob** [reply]
(Quite likely, the code in pybind11 can be copied almost verbatim)

**Reply by nschloe** [reply]
https://github.com/wjakob/nanobind/pull/299
